packages/workspace/src/browser/quick-open-workspace.ts

```diff
@@ -34,37 +35,40 @@ export class QuickOpenWorkspace implements QuickOpenModel {
     @inject(FileSystem) protected readonly fileSystem: FileSystem;
     @inject(LabelProvider) protected readonly labelProvider: LabelProvider;
     @inject(WorkspacePreferences) protected preferences: WorkspacePreferences;
+    @inject(EnvVariablesServer) protected readonly envServer: EnvVariablesServer;
 
     async open(workspaces: string[]): Promise<void> {
         this.items = [];
-        const homeStat = await this.fileSystem.getCurrentUserHome();
-        const home = (homeStat) ? new URI(homeStat.uri).path.toString() : undefined;
-        let tempWorkspaceFile: URI | undefined;
-        if (home) {
-            tempWorkspaceFile = getTemporaryWorkspaceFileUri(new URI(home));
-        }
-        await this.preferences.ready;
         if (!workspaces.length) {
             this.items.push(new QuickOpenGroupItem({
                 label: 'No Recent Workspaces',
                 run: (mode: QuickOpenMode): boolean => false
             }));
+            return;
         }
+
+        const [homeDirUri, configDirUri] = await Promise.all([
+            this.fileSystem.getCurrentUserHome(),
+            this.envServer.getConfigDirUri()
+        ]);
+        const homeDirPath = homeDirUri ? await this.fileSystem.getFsPath(homeDirUri.uri) : undefined;
+        const tempWorkspaceFile = getTemporaryWorkspaceFileUri(configDirUri);
+        await this.preferences.ready;
         for (const workspace of workspaces) {
             const uri = new URI(workspace);
             const stat = await this.fileSystem.getFileStat(workspace);
             if (!stat ||
                 !this.preferences['workspace.supportMultiRootWorkspace'] && !stat.isDirectory) {
                 continue; // skip the workspace files if multi root is not supported
             }
-            if (tempWorkspaceFile && uri.toString() === tempWorkspaceFile.toString()) {
+            if (uri.toString() === tempWorkspaceFile.toString()) {
                 continue; // skip the temporary workspace files
             }
             const icon = this.labelProvider.getIcon(stat);
             const iconClass = icon === '' ? undefined : icon + ' file-icon';
             this.items.push(new QuickOpenGroupItem({
                 label: uri.path.base,
-                description: (home) ? FileSystemUtils.tildifyPath(uri.path.toString(), home) : uri.path.toString(),
+                description: (homeDirPath) ? FileSystemUtils.tildifyPath(uri.path.toString(), homeDirPath) : uri.path.toString(),
```

Comment:
it will work for now, if we don't store data in homeDirPath anymore it will break.

Reply:
You are confused with what tilidyfing means, I had to look up too. It is always home holder relative on POSIX.

Reply:
ok